internal/tasks/device_render.go

```diff
@@ -66,29 +70,54 @@ func (t *DeviceRenderLogic) RenderDevice(ctx context.Context) error {
 	emptyIgnitionConfig, _, _ := config_latest.ParseCompatibleVersion([]byte("{\"ignition\": {\"version\": \"3.4.0\"}"))
 	t.renderedConfig = &emptyIgnitionConfig
 
+	invalidConfigs := []string{}
 	if device.Spec != nil && device.Spec.Config != nil {
 		for i := range *device.Spec.Config {
 			configItem := (*device.Spec.Config)[i]
-			err := t.handleConfigItem(ctx, &configItem)
+			name, err := t.handleConfigItem(ctx, &configItem)
 			if err != nil {
-				return t.setStatus(ctx, err)
+				invalidConfigs = append(invalidConfigs, name)
 			}
 		}
 	}
 
+	repoRefsErr := t.store.Device().OverwriteRepositoryRefs(ctx, t.resourceRef.OrgID, *device.Metadata.Name, t.repoNames...)
+
+	if len(invalidConfigs) != 0 {
+		configurationStr := "configuration"
+		if len(invalidConfigs) > 1 {
+			configurationStr += "s"
+		}
+		err = fmt.Errorf("device has %d invalid %s: %s", len(invalidConfigs), configurationStr, strings.Join(invalidConfigs, ", "))
+		return t.setStatus(ctx, err)
+	}
+
+	if repoRefsErr != nil {
+		return t.setStatus(ctx, repoRefsErr)
+	}
+
 	configjson, err := json.Marshal(t.renderedConfig)
 	if err != nil {
 		return t.setStatus(ctx, err)
 	}
 
+	if device.Metadata.Owner == nil || *device.Metadata.Owner == "" {
+		// The standalone device is valid, set the many-to-may relationship with the repos
+		err = t.store.Device().OverwriteRepositoryRefs(ctx, t.resourceRef.OrgID, *device.Metadata.Name, t.repoNames...)
```

Comment:
If Repository is an associated object that is part of Device would it make sense to update the Interface to allow something like this. Instead of bolting Repository methods into device directly. This can be a followup.

```suggestion
		err = t.store.Device(ctx, name).Repositories().Clear(ctx, t.repoNames...)
```